client/apps/edit/components/layout/index.coffee

```diff
@@ -133,3 +136,50 @@ module.exports = class EditLayout extends Backbone.View
   toggleDragover: (e) ->
     $(e.currentTarget).closest('.dashed-file-upload-container')
       .toggleClass 'is-dragover'
+
+  getLinkableText: ->
+    fullText = @getBodyText()
+    fullText.match(/==(\S[^==]*\S)==/ig)
+
+  autolinkText: ->
+    $('#autolink-status').addClass('searching').html('Linking...')
+    $('#edit-content__overlay').addClass('disabled')
+    linkableText = @getLinkableText()
+    async.mapLimit linkableText, 5, ((findText, cb) =>
+      text = findText.split('==').join('')
+      request
+        .get("/api/search?term=#{text}")
+        .set('X-Access-Token': sd.USER?.access_token)
+        .end (err, res) =>
+          if err or res.body.total < 1
+            @article.replaceLink(findText, text)
+            return cb()
+
+          valid_results = @findValidResults(res.body.hits)
+          if valid_results.length == 0
+            @article.replaceLink(findText, text)
+            return cb()
+
+          @article.replaceLink(findText, @getNewLinkFromHits(valid_results) || text)
+          return cb()
+    ), (err, result) =>
+      console.log 'herererere'
+      @article.sections.trigger 'change:autolink'
+      $('#autolink-status').removeClass('searching').html('Auto-Link')
+      $('#edit-content__overlay').removeClass('disabled')
+
+  findValidResults: (hits) ->
+    _.reject(hits, (h) -> h._score < 6.5 || h._source.visible_to_public == false)
+
+  getNewLinkFromHits: (results) ->
+    result = results[0]
+    name = result._source.name
+    link = @findLinkFromResult(result)
+    "<a href='#{link}'>#{name}</a>"
+
+  findLinkFromResult: (result) ->
+    switch result._type
+      when "artist" then "https://artsy.net/artist/#{result._source.slug}"
+      when "partner" then "https://artsy.net/#{result._source.slug}"
+      when "city" then "https://artsy.net/shows/#{result._source.slug}"
```

Comment:
we can probably get `https://artsy.net/shows` to a config if we don't have it as a config. We can do that in separate PR though. when i added that i wasn't sure if there is config for it.

Reply:
Oh true, "https://artsy.net/" can be replaced with the already existing sd.FORCE_URL

Reply:
Im doing this in a separate PR